src/templates/components/CommunityTemplateReadme.tsx

```diff
@@ -1,37 +1,39 @@
 // Libraries
-import React, {PureComponent} from 'react'
+import React, {Component} from 'react'
+import {connect, ConnectedProps} from 'react-redux'
 import {MarkdownRenderer} from 'src/shared/components/views/MarkdownRenderer'
-import {fetchReadMe} from 'src/templates/api'
-import {reportError} from 'src/shared/utils/errors'
 
-interface Props {
-  directory: string
+import {AppState} from 'src/types'
+import {getTemplateNameFromUrl} from 'src/templates/utils'
+import {fetchAndSetReadme} from 'src/templates/actions/thunks'
+
+interface OwnProps {
+  url: string
 }
 
+type ReduxProps = ConnectedProps<typeof connector>
+type Props = OwnProps & ReduxProps
+
 const cloudImageRenderer = (): any =>
   "We don't support images in markdown for security purposes"
 
-export class CommunityTemplateReadme extends PureComponent<Props> {
-  state = {readMeData: ''}
-  componentDidMount = async () => {
-    try {
-      const response = await fetchReadMe(this.props.directory)
-      this.setState({readMeData: response})
-    } catch (error) {
-      reportError(error, {
-        name: 'The community template fetch github readme failed',
-      })
-
-      this.setState({
-        readMeData: "## We can't find the readme associated with this template",
-      })
+class CommunityTemplateReadmeUnconnected extends Component<Props> {
+  componentDidMount = () => {
+    if (!this.props.communityTemplateReadmeCollection) {
```

Comment:
ahh, i told you to change it everywhere, but this is a place where it should be called something else. see below.